ukis_pysat/download.py

```diff
@@ -382,6 +346,156 @@ def __exit__(self, exc_type, exc_val, exc_tb):
+        return {
+            "id": self.id,
+            "platformname": None if self.platformname is None else self.platformname.value,
+            "producttype": self.producttype,
+            "orbitdirection": self.orbitdirection,
+            "orbitnumber": self.orbitnumber,
+            "relativeorbitnumber": self.relativeorbitnumber,
+            "acquisitiondate": None if self.acquisitiondate is None else self.acquisitiondate.strftime("%Y/%m/%d"),
+            "ingestiondate": None if self.ingestiondate is None else self.ingestiondate.strftime("%Y/%m/%d"),
+            "processingdate": None if self.processingdate is None else self.processingdate.strftime("%Y/%m/%d"),
+            "processingsteps": self.processingsteps,
+            "processingversion": self.processingversion,
+            "bandlist": self.bandlist,
+            "cloudcoverpercentage": self.cloudcoverpercentage,
+            "format": self.format,
+            "size": self.size,
+            "srcid": self.srcid,
+            "srcurl": self.srcurl,
+            "srcuuid": self.srcuuid,
+        }
+
+    def to_json(self):
+        """Converts Metadata to JSON.
+
+        :returns: Metadata (JSON)
+        """
+        return json.dumps(self.to_dict())
+
+    def to_geojson(self):
+        """Converts Metadata to GeoJSON.
+
+        :returns: Metadata (GeoJSON)
+        """
+        return geometry.mapping(_GeoInterface({"type": "Feature", "properties": self.to_dict(), "geometry": self.geom}))
+
+    def save(self, target_dir):
+        """Saves Metadata to GeoJSON file in target_dir with srcid as filename.
+
+        :param target_dir: Target directory that holds the downloaded metadata (String)
+        """
+        g = self.to_geojson()
+        with open(os.path.join(target_dir, g["properties"]["srcid"] + ".json"), "w") as f:
+            json.dump(g, f)
+
+
+class MetadataCollection:
+    """
+    Provides a container to store a collection of metadata objects. Conversion methods are provided to
+    analyse the metadata collection further.
+    """
+
+    __init__ = make_init()
+    items: List[Metadata] = field(type_hint=List[Metadata])
+
+    def to_dict(self):
+        """Converts MetadataCollection to List of Dict.
+
+        :returns: MetadataCollection (List of Dict)
+        """
+        return [item.to_dict() for item in self.items]
+
+    def to_json(self):
+        """Converts MetadataCollection to List of JSON.
+
+        :returns: MetadataCollection (List of JSON)
+        """
+        return [item.to_json() for item in self.items]
+
+    def to_geojson(self):
+        """Converts MetadataCollection to list of GeoJSON.
+
+        :returns: MetadataCollection (List of GeoJSON)
+        """
+        return [item.to_geojson() for item in self.items]
+
+    def to_pandas(self):
+        """Converts MetadataCollection to Pandas Dataframe.
+
+        :returns: MetadataCollection (Pandas Dataframe)
+        """
+        try:
+            import pandas as pd

```

Comment:
I hope that's okay, I thought for these convenience functions we don't need to import such a heavy library. Did the same for Dask.

Reply:
I agree and that's a good way to solve it.